src/main/java/uk/gov/hmcts/befta/DefaultTestAutomationAdapter.java

```diff
@@ -30,19 +31,43 @@
     private final AuthApi idamApi;
     private final ServiceAuthorisationApi serviceAuthorisationApi;
 
-    private final Map<String, ServiceAuthTokenGenerator> tokenGenerators = new ConcurrentHashMap<>();
-
-    private final Map<String, UserData> users = new HashMap<>();
+    private Cache<String, UserData> users;
+    private Cache<String, ServiceAuthTokenGenerator> tokenGenerators;
 
     private BeftaTestDataLoader dataLoader;
 
     public DefaultTestAutomationAdapter() {
         serviceAuthorisationApi = BeftaServiceAuthorisationApiClientFactory.createServiceAuthorisationApiClient();
         idamApi = BeftaIdamApiClientFactory.createAuthorizationClient();
+
+        dataLoader = buildTestDataLoader();
+
+        setupUsersCache();
+        setupTokenGeneratorsCache();
+    }
+
+    private void setupUsersCache() {
+        CacheBuilder<Object, Object> userCacheBuilder = CacheBuilder.newBuilder();
+
+        Long userCacheTtl = BeftaMain.getConfig().getUserTokenCacheTtlInSeconds();
+        if (userCacheTtl != 0) {
+            userCacheBuilder.expireAfterAccess(userCacheTtl, TimeUnit.SECONDS);
+        }
+        users = userCacheBuilder.build();
```

Comment:
`userCacheBuilder.recordStats()build();` would allow following stats relating to the cache to be recorded

https://guava.dev/releases/snapshot/api/docs/com/google/common/cache/CacheStats.html